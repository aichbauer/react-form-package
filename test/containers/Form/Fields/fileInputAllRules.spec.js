import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

import {
  Form,
  Field,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with file input | file input all rules | preOnChange -> should take files not value', () => {
  const event = {
    target: {
      id: 'file',
      type: 'file',
      value: 'value',
      files: {
        0: 'i am a file trust me',
      },
    },
  };
  const rules = {
    type: 'file',
  };
  const options = {
    preOnChange: () => ({ 0: 'compressed file' }),
  };

  const myComponent = (
    <Form>
      <div>
        <div>email</div>
        <Field
          type="file"
          id="file"
        />
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();

  const wrapper = shallow(<Form />);
  const instance = wrapper.instance();

  instance.setInitialState([
    <Field
      type="file"
      id="file"
    />,
  ]);
  wrapper.update();

  instance.handleOnChange(event, rules, options);
  wrapper.update();

  const state = wrapper.state();

  expect(state.data.file.value).toBe('value');
  expect(state.data.file.files).toEqual({ 0: 'compressed file' });

  expect(tree).toMatchSnapshot();
});
