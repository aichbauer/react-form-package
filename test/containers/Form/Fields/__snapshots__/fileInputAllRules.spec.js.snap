// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with file input | file input all rules | preOnChange -> should take files not value 1`] = `
<form
  enctype="application/x-www-form-urlencoded"
>
  <div>
    <div>
      email
    </div>
    <input
      className="rfp-input"
      id="file"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="file"
      valid={false}
      value=""
    />
  </div>
</form>
`;
